### PR TITLE
RFC: switch to using the full oraclelinux:8 base image for the Oracle Linux Developer images

### PR DESCRIPTION
**Pull request by Djelibeybi**
This makes the developer images far more developer-friendly by including such useful tools as a module-aware `dnf` and more. Yes, the images jump in size, but then developer images are inherently larger because of their purpose.

Of particular importance is for folks to test the builds at https://github.com/Djelibeybi?tab=packages to make sure current functionality is unaffected, including any downstream CI/CD tooling that assumes the existence of `microdnf`.

### CODE REVIEW

**Review by crush-157 — APPROVED** [review]
The `2.7-node` image seems fine.  I've hit an issue with compiling native gem extensions in the `2.7` image, but it's consistent with the current (official) image, so I'll look at this once we've fattened the images.

**Comment by Djelibeybi** [comment]
> I've hit an issue with compiling native gem extensions in the `2.7` image, but it's consistent with the current (official) image, so I'll look at this once we've fattened the images.

That would be great.

**Comment by AmedeeBulle** [comment]
Can you clarify how you would like to transition to `oraclelinux:8` beyond this RFC?

Is the plan to re-base all dev images on the `compat` image and as second step have the code owners move from `microdnf` to `dnf` and drop the dependency on `compat`?

For the _migrated_ images (like your `golan/latest`)  shouldn't we drop the module file?

**Comment by Djelibeybi** [comment]
No, we have to maintain `microdnf` and the module files until at least the next Oracle Linux update is released, to give downstream consumers of the images time to update their tooling. My plan is actually to provide two updates cycles for this. 

When the next update is released, my plan is to replace `microdnf` with a script that outputs a notice saying the image has changed and to switch to `dnf` before the next update is released, at which point `microdnf` will be removed from this image. Then the script will pause for 30 seconds. That pause should be sufficient for folks to notice their CI jobs are taking longer and to check the logs. 

At the next update, we can finalise the switch by removing the module files and switching directly to the `oraclelinux:8` base image.

File: OracleLinuxDevelopers/oraclelinux8/nodejs/14-oracledb/Dockerfile
Comment (cjbj) [inline OracleLinuxDevelopers/oraclelinux8/nodejs/14-oracledb/Dockerfile:10]: The IC Basic package has a dependency on libaio, so technically this explicit dependency isn't need.

File: OracleLinuxDevelopers/oraclelinux8/golang/latest/Dockerfile
Comment (cjbj) [inline OracleLinuxDevelopers/oraclelinux8/golang/latest/Dockerfile:3]: To end users this 'strange' repo would looks sus.  Can you add some comments to each Dockerfile to explain what it is, and what the plan is?

**Review by cjbj — CHANGES_REQUESTED** [review]

File: OracleLinuxDevelopers/oraclelinux8/golang/latest/Dockerfile
Comment (Djelibeybi) [reply OracleLinuxDevelopers/oraclelinux8/golang/latest/Dockerfile:3]: Whoops, that was for testing purposes. I need to merge the compat image into this repo and change the URL. It will be `ghcr.io/oracle/oraclelinux8-compat:8-slim` eventually.

**Comment by Djelibeybi** [comment]
Once #2044 is merged and the `oraclelinux8-compat:8-slim` image is built, I will update this PR to use that image from Oracle's GHCR.

**Comment by Djelibeybi** [comment]
@AmedeeBulle I would deeply appreciate it if you could fix the workflow breakage I caused. :)

File: OracleLinuxDevelopers/oraclelinux8/nodejs/14-oracledb/Dockerfile
Comment (Djelibeybi) [reply OracleLinuxDevelopers/oraclelinux8/nodejs/14-oracledb/Dockerfile:10]: I've left the original Dockerfiles intact, i.e. I just want to change the base image. We can fix this afterwords.

**Comment by Djelibeybi** [comment]
@AmedeeBulle @mark-au @totalamateurhour can I please get another review/approval from one of you? I'd like to merge this sooner rather than later.

File: OracleLinuxDevelopers/oraclelinux8/nodejs/14-oracledb/Dockerfile
Comment (cjbj) [inline OracleLinuxDevelopers/oraclelinux8/nodejs/14-oracledb/Dockerfile:3]: Stray backquote

**Review by cjbj — DISMISSED** [review]
I checked the two with DB drivers; they build and run fine.

- Can/should you augment the slim note to say whether users should migrate from microdnf?
- Technically you can remove the explicit install of oracle-instantclient-basic since that gets pulled in automatically.  I'm in two minds about 'hiding' this, since the result is the same.
- There is at least one changed file with only a 2020 copyright.

**Comment by Djelibeybi** [comment]
Thanks @cjbj, I've pushed changes to the main `README.md` with a deprecation notice for `microdnf` and I've updated all the applicable copyright notices. I'm going to leave the explicit installation of `oracle-instantclient-basic` because there's no change in the result and it makes it more obvious that it's part of the image when only reviewing the source.

**Comment by Djelibeybi** [comment]
@AmedeeBulle @totalamateurhour @mark-au it would be great to get another review, but I will be merging this change in regardless by the end of the week, because we need to stop using `8-slim`.

**Review by AmedeeBulle — APPROVED** [review]

**Pull request merged**
